feat(knowledge): Реализована секция knowledge

VeronikaBulaeva opened the pull request.

VeronikaBulaeva committed:
feat(knowledge): Реализована секция knowledge

VeronikaBulaeva committed:
feat(knowledge): Реализована секция knowledge

VeronikaBulaeva committed:
feat(knowledge): Реализована секция knowledge

majishanred commented:
Я бы в компонентах сделал что-то типа папки commons для тех компонентов, которые много где появляются типа кнопки и вынес ее туда. Но надо остальных спросить

JstUsername commented on src/theme.tsx:
```diff
@@ -2,6 +2,15 @@ import '@fontsource/roboto';
 import { createTheme } from '@mui/material/styles';
 
 export const theme = createTheme({
+  breakpoints: {
+    values: {
+      xs: 0,
+      sm: 600,
+      md: 900,
+      lg: 1280,
+      xl: 1440,
+    },
+  },
```
Из задания:
Mobile 360px
Tablet 768px
Ipad 1024px
Desktop 1440px
2k 1920px
4k 3240px

VeronikaBulaeva replied:
Окей, но xs должна быть 0, чтобы на экранах меньше 360 тоже ок было и вёрстка не летела

JstUsername replied:
А есть риски что полетит?
У меня прост есть медиа запросы на то что ниже sm, если сделать 0, то не будут срабатывать.
Ну или тогда в тему делать 0, а текущий sm я руками пропишу

JstUsername replied:
А, нет, все норм, у себя тогда тоже сейчас xs на 0 сделаю

JstUsername committed:
feat(assets/login): добавлено изображение и иконки для компонента Login

JstUsername committed:
feat(Login.tsx): добавлен компонент регистрации и авторизации

JstUsername committed:
feat(Login.tsx): добавлен компонент регистрации и авторизации

JstUsername committed:
feat(Login): компонент регистрации и авторизации добавлен в App.tsx

JstUsername commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,72 @@
+        </KnowledgeText>
+      </Box>
+      <Flex>
+        <DefaultButton type={ButtonType.contained} text={'Для врачей'} />
+        <DefaultButton type={ButtonType.outlined} text={'Для пациентов'} />
```
"Дергаются" кнопки при наведении

JstUsername replied:
Размер шрифта для мобильной версии 14, в этом блоке.
В общий тип наверное не стоит добавлять, они там почти везде разные

JstUsername commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,72 @@
+            </Box>
+          </Grid>
+          <Grid item container spacing={3} xs={6} alignSelf={'center'} flex={1}>
+            {knowledgeInfo.map(({ statistics, image, description, id }) => (
+              <KnowledgeInfo
+                key={`knowledgeInfo${id}`}
+                image={image}
+                statistics={statistics}
+                description={description}
+              />
```
График крутой

JstUsername commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,72 @@
+
+const Knowledge: FC = () => {
+  return (
+    <Container maxWidth={'xl'} sx={{ background: '#FFF', marginBlock: 10 }}>
```
В мобильной версии margin: 48px 🤓

JstUsername committed:
fix(StyledOutlinedButton): удалены лишние значения padding

JstUsername commented:
Не забудь удалить .gitkeep перед слиянием

VeronikaBulaeva committed:
fix(knowledge): Поправлены комментарии

VeronikaBulaeva committed:
fix(knowledge): Поправлены комментарии

VeronikaBulaeva committed:
fix(knowledge): Поправлены комментарии

JstUsername committed:
feat(generics): удалены примитивные типы для стейтов

u1timaD commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -3,13 +3,13 @@ import { Box, Container, Grid, styled, Typography } from '@mui/material';
 
 const Knowledge: FC = () => {
   return (
-    <Container maxWidth={'xl'} sx={{ background: '#FFF', marginBlock: 10 }}>
```
по семантике тут должен быть блок section

u1timaD replied:
мне кажется тут боковые отступы должны быть падинги от внешнего контейнера. Короче)) мне кажется не хватает обёртки на три дива

u1timaD commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -3,13 +3,13 @@ import { Box, Container, Grid, styled, Typography } from '@mui/material';
   return (
-    <Container maxWidth={'xl'} sx={{ background: '#FFF', marginBlock: 10 }}>
+    <Container maxWidth={'lg'} sx={{ background: '#FFF', marginBlock: { xs: 5, sm: 10 } }}>
       <Box mb={6}>
```
маршжины можно заменить, оберунтть все три дива , родителю флекс и gap 48px, вместо того , чтобы каждому задавать mb

u1timaD commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -20,23 +20,23 @@ const Knowledge: FC = () => {
-              maxWidth={{ sx: 630, mb: 900, lg: 630 }}
+              maxWidth={{ xs: 768, sm: 1024, md: 1440, lg: 630 }}
             >
               <Chart />
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/4f0131dd-6a64-4ca6-a568-5d8aa905f7bf)

больше чем в макете

u1timaD commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -20,23 +20,23 @@ const Knowledge: FC = () => {
             </Box>
           </Grid>
-          <Grid item container spacing={3} xs={6} alignSelf={'center'} flex={1}>
+          <Grid item container xs={6} spacing={3} alignSelf={'center'} justifyContent={'space-between'} flex={1}>
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/7b0feef8-3180-4b61-88b2-458f4d14477a)
зачем тут падинг.. не понял

VeronikaBulaeva replied:
если посмотришь в доке, так работает Grid и spacing в MUI, в коде нет паддингов

IvanKoigerov replied:
можешь убрать их тогда если их нет в макете @VeronikaBulaeva

u1timaD commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -20,23 +20,23 @@ const Knowledge: FC = () => {
             </Box>
           </Grid>
-          <Grid item container spacing={3} xs={6} alignSelf={'center'} flex={1}>
+          <Grid item container xs={6} spacing={3} alignSelf={'center'} justifyContent={'space-between'} flex={1}>
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/3816a6e6-a910-4e34-85ca-799dda1315ed)
тут тоже не понял, у тебя же grid, убери падинги сделать gap

VeronikaBulaeva replied:
тоже самое, что и с другим паддингом

u1timaD commented on src/components/Knowledge/knowledge-info/KnowledgeInfo.tsx:
```diff
@@ -0,0 +1,31 @@
+          alignItems={{ xs: 'center', lg: 'flex-start' }}
+        >
+          <Grid item alignItems={'center'}>
+            <img src={image} alt="" />
```
альты должны быть всегда и размер картинки у тебя 49х48 
а должен быть 48х48
![image](https://github.com/JstUsername/team-landing/assets/45046356/c436a0a4-3062-4d9b-b063-9c8afd52e295)

u1timaD commented on src/components/Knowledge/knowledge-info/KnowledgeInfo.tsx:
```diff
@@ -0,0 +1,31 @@
+          columnSpacing={2}
+          alignItems={{ xs: 'center', lg: 'flex-start' }}
+        >
+          <Grid item alignItems={'center'}>
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/7bbf1485-38cc-4fcd-a67a-0156dadaa753)
там у тебя должны работать внешние падинги, а они схлопываются)

VeronikaBulaeva replied:
в коде паддингов нет, так работает Grid в MUI

u1timaD commented on src/App.tsx:
```diff
@@ -1,13 +1,14 @@
+        <Knowledge />
       </div>
     </ThemeProvider>
   );
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/dd261df7-1650-4661-aa0b-d3f5d9469c0e)
ошибка в консоли

VeronikaBulaeva committed:
fix(knowledge): Поправлены комментарии

JstUsername committed:
fix(props): удалены лишние props при обращении к теме

JstUsername committed:
fix(handleClickShowPassword): setShowPassword вынесен в onClick

JstUsername committed:
feat(handleKeyDown): добавлена функция для переключения чекбокса по нажатию на клавиши

JstUsername committed:
feat(StyledLink): добавлены стили для Link исправляющие ширину

JstUsername committed:
fix(InputLabel): заголовки для input реализованы с помощью label

JstUsername committed:
fix(letter-spacing): удалены свойства letter-spacing

JstUsername commented on src/components/Knowledge/Knowledge.tsx:
Почему кстати используется .tsx? Тут же вроде и обычного .ts будет достаточно.

JstUsername commented on src/constants/constants.tsx:
Почему кстати используется .tsx? Тут же вроде и обычного .ts будет достаточно.

VeronikaBulaeva replied:
потому что используем svgr, так что теперь svg передаем как реакт компонент

JstUsername replied:
Для этого вроде все равно .tsx не нужен, подожду коммита, изменю у себя если что.
По - идее там передача svg как FunctionComponent<SVGProps<SVGSVGElement>>.
Получается что - то типо:
```
import MenuIcon from '../assets/header/icons/menu.svg?react';
export const BottomHeaderItems: HeaderLi[] = [
  {
    label: 'Меню',
    active: true,
    icon: MenuIcon,
  },
  ]
```

JstUsername replied:
В принципе вопрос исчерпан, все равно отдельные константы у всех будут ). Спрашивал чтобы при слиянии конфликтов не было.

IvanKoigerov commented on src/commons/button/DefaultButton.tsx:
```diff
@@ -0,0 +1,34 @@
+  );
+};
+
+const Button = styled(MuiButton)((props: { fontsizesx: number } & ButtonProps) => ({
+  'paddingInline': 32,
+  'paddingBlock': 16,
+  'border': ['2', 'solid'],
+  'borderRadius': 0,
+  'borderColor': theme.palette.info.main,
+  'boxShadow': 'none',
+  'textTransform': 'none',
+  ':hover': {
+    border: ['2', 'solid'],
+    borderColor: '#3064ac',
+  },
+  '@media (max-width: 768px)': {
+    paddingInline: 28,
+    paddingBlock: 14,
+    fontSize: props.fontsizesx,
+  },
+}));
```
Стили пишем в одном фиде в формате 
```
Item = styled("div")`

`
```

majishanred commented on src/commons/button/DefaultButton.tsx:
```diff
@@ -0,0 +1,34 @@
+import { IDefaultButton } from './types';
+import { theme } from '../../theme';
+
+const DefaultButton: FC<IDefaultButton> = ({ type, text, fontsizesx }) => {
+  return (
+    <Button variant={type} color={'info'} fontsizesx={fontsizesx}>
+      {text}
+    </Button>
+  );
+};
```
Довольно странно в кноку прокидывать текст таким образом, переделай на children.

majishanred commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,83 @@
+          </KnowledgeText>
+        </Box>
+        <FlexButton>
+          <DefaultButton type={ButtonType.contained} text={'Для врачей'} fontsizesx={14} />
+          <DefaultButton type={ButtonType.outlined} text={'Для пациентов'} fontsizesx={14} />
```
А ведь по факту это не кнопки, а ссылки должны быть. У меня такой же косяк есть, надо править.

majishanred commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,83 @@
+  'fontWeight': 700,
+  'lineHeight': 1.1,
+  'letterSpacing': '0em',
+  '@media (max-width: 768px)': {
+    fontSize: 24,
+  },
```
Можно ссылаться на брейкпоинты в теме.

VeronikaBulaeva committed:
fix(knowledge):

Добавлен svgr, исправлено написание стилей, исправлены комментарии

VeronikaBulaeva committed:
fix(knowledge):

Добавлен svgr, исправлено написание стилей, исправлены комментарии

VeronikaBulaeva committed:
fix(knowledge):

Добавлен svgr, исправлено написание стилей, исправлены комментарии

JstUsername committed:
feat(svgr): добавлен и настроен плагин svgr

majishanred commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,106 @@
+        </FlexButton>
+        <Box sx={{ width: '100%' }}>
+          <FlexKnowledgeItem>
+            <Box
+              border={1}
+              borderColor={'divider'}
+              mb={{ xs: 3, lg: 0 }}
+              minHeight={396}
+              maxHeight={396}
+              maxWidth={{ xs: 768, sm: 1024, md: 1440, lg: 535 }}
+            >
```
Многовато пропсов, принято что если их такое большое кол-во то в отдельный компонент выносить надо. Но это надо комента от Вани или Дена подождать.

myaak replied:
В стайлед нужно бы вынести

majishanred replied:
+

majishanred commented on src/commons/button/DefaultButton.tsx:
```diff
@@ -0,0 +1,31 @@
+import { IDefaultButton } from './types';
+import { theme } from '../../theme';
+
+const DefaultButton: FC<PropsWithChildren<IDefaultButton>> = ({ type, children }) => {
+  return (
+    <Button component={Link} variant={type} color={'info'}>
+      {children}
+    </Button>
+  );
+};
```
Может быть вкусовщина, но тогда компонент лучше переименовать в ContainedLink. Потому что захочет кто-нибудь дефолтную кнопку юзнуть зайдет в тулзы и увидит линк. И пойдет возьня откуда это.

myaak replied:
Плохо, Button используется муишный, это должен быть ui компонент основанный на MUI тогда.
Вы должны расширять тип уже существующих пропов из MUI. На примере тут ButtonProps
И на основе него уже прокидывать в этот компонент. Он не похож на универсальный.

Если я захочу использовать эту кнопку в компоненте, но цвет мне нужен не 'info', мне пойти что ли кнопку другую сделать? В худшем случае можно сделать дефолтные значения, если не хотите прописывать вариант и цвет, но я должен кастомизировать эту кнопку как МУИшную, как захочу, не трогая логику этого компонента.
UPD: если не поймете, то мб подскажу как сделать.

VeronikaBulaeva replied:
Так у меня сейчас как раз Button основан на MUI, просто называется также

myaak replied:
В этом проблем нет, тут все хорошо, просто не универсально. Описал выше

VeronikaBulaeva committed:
fix(swiper):

Исправлены комментарии

VeronikaBulaeva committed:
fix(swiper):

Исправлены комментарии

VeronikaBulaeva committed:
fix(swiper):

Исправлены комментарии

myaak commented on src/commons/button/DefaultButton.tsx:
```diff
@@ -0,0 +1,31 @@
+import { FC, PropsWithChildren } from 'react';
+import { ButtonProps, Link, styled } from '@mui/material';
+import MuiButton from '@mui/material/Button';
```
Такие вещи обычно из корня MUI испортят, только в виде
`import {Button as MUIButton} from '@mui/material';`

VeronikaBulaeva committed:
fix(swiper):

Исправлены комментарии

JstUsername committed:
fix(LoginProps): исправлено название интерфейса

JstUsername committed:
fix(theme): исправлено использование theme в styled-components

JstUsername committed:
fix(Login.tsx): компонент Login вынесен в отдельную папку

JstUsername committed:
fix(Login, App): исправлены импорты после переноса Login

JstUsername committed:
feat(Input.tsx): поля формы и их стили вынесены в отдельный компонент

JstUsername committed:
fix(Grid): стилизация Grid вынесена в styled-components

JstUsername committed:
feat(dependencies): в проект добавлен React Router

JstUsername committed:
feat(BrowserRouter): компонент App обернут в BrowserRouter

JstUsername committed:
feat(Routes, AppContent): добавлены пути, контент приложения вынесен в AppContent

JstUsername committed:
fix(routing): роутинг реализован с помощью React Router

IvanKoigerov commented on src/commons/button/ButtonLink.tsx:
```diff
@@ -0,0 +1,30 @@
+  :hover {
+    border: 2px solid #3064ac;
+  };
+  @media (max-width: 768px) {
```
theme.breakpoints.up и theme.breakpoints.down

IvanKoigerov replied:
Везде исправь плез

IvanKoigerov commented on src/commons/button/ButtonLink.tsx:
```diff
@@ -0,0 +1,30 @@
+import { FC, PropsWithChildren } from 'react';
+import { ButtonProps, Link, styled } from '@mui/material';
+import { Button as MUIButton } from '@mui/material';
+import { theme } from '../../theme';
```
Тему нельзя импортировать вне файле где она используется для themeProvider

IvanKoigerov commented on src/components/Knowledge/Knowledge.tsx:
```diff
@@ -0,0 +1,108 @@
+  gap: 16px;
+`;
+
+const KnowledgeText = styled(MuiTypography)`
+  font-size: 42px;
+  font-weight: 700;
+  letter-spacing: 0;
+  @media (max-width: ${({ theme }) => theme.breakpoints.values.sm}px) {
+    font-size: 24px;
+  },
+`;
```
Все что связано с текстом, то зачастую можно переиспользовать, ппоэтому выносим все эти значения в тему, создаем новый вариант

IvanKoigerov commented on src/components/Knowledge/chart/Chart.tsx:
```diff
@@ -0,0 +1,21 @@
+import { chartConfig, data } from '../constants';
+
+const Chart: FC = () => {
+  const plugin = {
+    id: 'increase-legend-spacing',
+    beforeInit(chart: any) {
+      const originalFit = chart.legend.fit;
+      chart.legend.fit = function fit() {
+        originalFit.bind(chart.legend)();
+        this.height += 16;
+      };
+    },
+  };
```
жестко

IvanKoigerov commented on src/components/Knowledge/constants.tsx:
Круто разобралась, большая молодец

IvanKoigerov commented on src/components/Knowledge/knowledge-info/KnowledgeInfo.tsx:
```diff
@@ -0,0 +1,42 @@
+
+const KnowledgeInfo: FC<Omit<IKnowledgeInfo, 'id'>> = ({ image, description, statistics }) => {
+  return (
+    <Box border={1} borderColor={'divider'} borderRadius={2} p={{ xs: 1, md: 2 }} minHeight={{ xs: 168, sm: 0 }}>
```
borderColor="divider"

IvanKoigerov commented on src/components/Knowledge/knowledge-info/KnowledgeInfo.tsx:
```diff
@@ -0,0 +1,42 @@
+import { Box, styled, Typography } from '@mui/material';
+import { IKnowledgeInfo } from '../types';
+
+const KnowledgeInfo: FC<Omit<IKnowledgeInfo, 'id'>> = ({ image, description, statistics }) => {
```
Тема с FC устарела, я бы не советовал бы ее использовать, а описывать так 
`({...}: Omit<IKnowledgeInfo, 'id'>)`

IvanKoigerov commented on src/components/Knowledge/knowledge-info/KnowledgeInfo.tsx:
```diff
@@ -0,0 +1,42 @@
+
+const GridContainer = styled('div')`
+  display: flex;
+  flex-direction: row;
```
row нет смысла указывать

VeronikaBulaeva committed:
fix(knowledge): media запросы заменены на theme.breakpoints.down

VeronikaBulaeva committed:
fix(knowledge): Поправлены комментарии

JstUsername committed:
feat(commons): компонент Input переименован в FormInput и вынесен в commons

JstUsername committed:
fix(Input): исправлен иморт, Input => FormInput

JstUsername committed:
fix(Login): компонент переименован в LoginPage

JstUsername committed:
fix(FormInput): исправлено название интерфейса и компонента

JstUsername committed:
feat(HomePage.tsx): добавлен компонент для размещения основного контента приложения

JstUsername committed:
fix(import, routing): исправлены импорты и пути для роутинга, AppContent заменен на HomePage

JstUsername committed:
fix(BrowserRouter): BrowserRouter вынесен из main в App

JstUsername committed:
fix(icons): иконки отображены с помощью svgr

JstUsername committed:
fix(login.webp): logn.webp импортирован как модуль, исправлен src

JstUsername committed:
fix(@media): все медиа запросы переписаны на theme.breakpoints.(down/up)

JstUsername committed:
fix(responsive h2): изменение размера h2, в зависимости от брейкпоинта sm вынесено в theme

JstUsername committed:
fix(StyledLink): StyledLink вынесен в компонент FormLink

JstUsername committed:
fix(StyledCheckbox): StyledCheckbox вынесен в компонент FormCheckbox

JstUsername committed:
feat(assets/login): добавлена иконка закрытого глаза

JstUsername committed:
feat(FormPasswordInput): инпут для пароля вынесен из LoginPage в отдельный компонент.

Для FormPasswordInput добавлена иконка закрытого глаза при видимом пароле.

JstUsername committed:
fix(EyeCloseIcon): увеличен размер EyeCloseIcon и добавлен отступ

JstUsername committed:
feat(validation): добавлена валидация email и пароля.

Email валидируется на входе и регистрации, пароль валидируеся только на регистрации.

JstUsername committed:
fix(theme): переопределена типизация темы, добавлен variant="formHeader"

IvanKoigerov commented:
Хорошо, можем сливать после Андрея

JstUsername committed:
fix(padding): исправлены отступы в FormInput и LoginPage

JstUsername committed:
fix(FormControl): FormControl вынесен в FormInput, для FormInput добавлен inputSx

JstUsername committed:
feat(utils/validate): функции для валидации email и пароль вынесены в utils/validate

JstUsername committed:
feat(FormEmailInput): добавлен отдельный компонент для email.

FormInput очищен от стейтов и условий, в FormPassword добавлен PasswordSubtitle.

JstUsername committed:
fix(file structure): страницы перемещены в pages, содержимое commons перемещено в components/form

JstUsername committed:
fix(import): исправлены импорты после перемещения commons, LoginPage, HomePage

VeronikaBulaeva committed:
fix(knowledge): Исправлено использование theme в кнопке

VeronikaBulaeva committed:
Merge remote-tracking branch 'origin/main' into feature/knowledge

# Conflicts:
#	package-lock.json
#	package.json
#	src/pages/HomePage/HomePage.tsx
#	vite.config.ts

VeronikaBulaeva committed:
fix(knowledge): Исправлено использование theme в кнопке
